service/worker/deployment/deployment_workflow.go

```diff
@@ -119,22 +119,26 @@ func (d *DeploymentWorkflowRunner) run() error {
 				pendingUpdates--
 			}()
 
+			// if no TaskQueueFamilies have been registered for the deployment
 			if d.DeploymentLocalState.TaskQueueFamilies == nil {
 				d.DeploymentLocalState.TaskQueueFamilies = make(map[string]*deployspb.DeploymentLocalState_TaskQueueFamilyInfo)
 			}
-			if d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName] == nil {
-				d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName] = &deployspb.DeploymentLocalState_TaskQueueFamilyInfo{}
+			// if no TaskQueues have been registered for the TaskQueueName
+			if d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName].TaskQueues == nil {
+				d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName].TaskQueues = make(map[int32]*deployspb.DeploymentLocalState_TaskQueueFamilyInfo_TaskQueueInfo)
 			}
 
-			// TODO Shivam - make UpdateHandler idempotent
+			// idempotency check
+			if _, ok := d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName].TaskQueues[int32(updateInput.TaskQueueType)]; ok {
+				return nil
+			}
 
 			// Add the task queue to the local state
 			newTaskQueueWorkerInfo := &deployspb.DeploymentLocalState_TaskQueueFamilyInfo_TaskQueueInfo{
 				TaskQueueType:   updateInput.TaskQueueType,
```

Comment:
Nit: is this field needed anymore with the map now?

Reply:
you are right in the sense we can just take it from the key but I think it's neat to always fetch values from values in a map rather than key (assuming we need task-queue type for some reason)

I say we keep it for now and replace it later on if not required